CMakeLists.txt

```diff
@@ -30,19 +34,23 @@ function(packageProject)
     add_library(${PROJECT_NAMESPACE}${PROJECT_NAME} ALIAS ${PROJECT_NAME})
   endif()
 
-  if (DEFINED PROJECT_VERSION_HEADER)
+  if(DEFINED PROJECT_VERSION_HEADER)
     set(PROJECT_VERSION_INCLUDE_DIR ${PROJECT_BINARY_DIR}/PackageProjectInclude)
     string(TOUPPER ${PROJECT_NAME} UPPERCASE_PROJECT_NAME)
     configure_file(
       ${PACKAGE_PROJECT_ROOT_PATH}/version.h.in
       ${PROJECT_VERSION_INCLUDE_DIR}/${PROJECT_VERSION_HEADER}
       @ONLY
     )
+
+    # TODO: not clear to me why? CK
     get_target_property(target_type ${PROJECT_NAME} TYPE)
-    if (target_type STREQUAL "INTERFACE_LIBRARY")
+    if(target_type STREQUAL "INTERFACE_LIBRARY")
       set(VISIBILITY INTERFACE)
+      message(STATUS "target ${PROJECT_NAME} is INTERFACE_LIBRARY")
     else()
-      set(VISIBILITY PUBLIC)
+      set(VISIBILITY PUBLIC)    # TODO WHY not PRIVATE? Check this! CK
```

Comment:
If it were private, then the installed headers could not be found?

Reply:
OK, it is needed for standalone project, right?